ovirt_imageio/client/_tool.py

```diff
@@ -27,7 +34,23 @@ def main():
         format="%(asctime)s %(levelname)-7s (%(threadName)s) [%(name)s] "
                "%(message)s")
 
-    # XXX Setup signal handlers.
-    # XXX Handle commands errors.
-
-    args.command(args)
+    signal.signal(signal.SIGINT, _handle_signal)
+    signal.signal(signal.SIGTERM, _handle_signal)
+
+    try:
+        args.command(args)
+    except Exception:
+        if terminated:
```

Comment:
As it is right now, terminated will never be `0`, so this is fine, but technically it should be `if terminated is not None:`. Similar but opposite in the handler.

Reply:
Thinking again, since signal cannot be 0, using terminated as boolean is much more clear and better then using "is not None".